c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -790,6 +791,15 @@ cudaError_t cudaMallocMaybeCapturing(void** p, size_t size) {
 
 static std::string reportProcessMemoryInfo(int device) {
 #ifdef PYTORCH_C10_DRIVER_API_SUPPORTED
+  void* handle_1 = dlopen("libnvidia-ml.so.1", RTLD_LAZY);
+  try {
+    TORCH_INTERNAL_ASSERT(handle_1);
+  } catch (...) {
```

Comment:
Similar to below, can we avoid a catch-all?

Reply:
Followed your offline suggestion to change this to a simple if check, thanks.